lifecycle-operator/webhooks/pod_mutator/handlers/appcreationrequest_handler.go

```diff
@@ -0,0 +1,97 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	klcv1alpha3 "github.com/keptn/lifecycle-toolkit/lifecycle-operator/apis/lifecycle/v1alpha3"
+	apicommon "github.com/keptn/lifecycle-toolkit/lifecycle-operator/apis/lifecycle/v1alpha3/common"
+	controllercommon "github.com/keptn/lifecycle-toolkit/lifecycle-operator/controllers/common"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AppCreationRequestHandler struct {
+	Client      client.Client
+	Log         logr.Logger
+	Tracer      trace.Tracer
+	EventSender controllercommon.IEvent
+}
+
+func (a *AppCreationRequestHandler) Handle(ctx context.Context, pod *corev1.Pod, namespace string) error {
+
+	ctx, span := a.Tracer.Start(ctx, "create_app", trace.WithSpanKind(trace.SpanKindProducer))
+	defer span.End()
+
+	newAppCreationRequest := generateAppCreationRequest(ctx, pod, namespace)
+	newAppCreationRequest.SetSpanAttributes(span)
+
+	a.Log.Info("Searching for AppCreationRequest", "appCreationRequest", newAppCreationRequest.Name, "namespace:", newAppCreationRequest.Namespace)
+
+	appCreationRequest := &klcv1alpha3.KeptnAppCreationRequest{}
+	err := a.Client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: newAppCreationRequest.Name}, appCreationRequest)
+	if errors.IsNotFound(err) {
+		return a.createApp(ctx, newAppCreationRequest, span)
+	}
+
+	if err != nil {
+		span.SetStatus(codes.Error, err.Error())
+		return fmt.Errorf("could not fetch AppCreationRequest"+": %+v", err)
+	}
+	a.Log.Info("Found AppCreationRequest", "appCreationRequest", newAppCreationRequest.Name, "namespace:", newAppCreationRequest.Namespace)
+	return nil
+}
+
+func (a *AppCreationRequestHandler) createApp(ctx context.Context, newAppCreationRequest *klcv1alpha3.KeptnAppCreationRequest, span trace.Span) error {
```

Comment:
```suggestion
func (a *AppCreationRequestHandler) createAppCreationRequest(ctx context.Context, newAppCreationRequest *klcv1alpha3.KeptnAppCreationRequest, span trace.Span) error {
```

Alternatively, we could rename this to a more generic `createResource` (and `generateResource` for the other function in this file), since the name `AppCreationRequestHandler` already implicates which kind of resource we are working with